dealgenie_scoring_engine_v4_3_toc.py

```
@@ -0,0 +1,498 @@
+        # Add TOC data to dataframe
+        toc_df = pd.DataFrame(toc_results)
+        for col in toc_df.columns:
+            df[f'toc_{col}'] = toc_df[col]
+        
+        print(f"✓ TOC data added to {len(df)} properties")
+        print(f"  Properties with valid coordinates: {valid_coords}")
+        
+        # TOC tier distribution
+        if valid_coords > 0:
+            tier_dist = df['toc_toc_tier'].value_counts().sort_index()
+            print(f"\nTOC Tier Distribution:")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)